docs-src/modules/ROOT/pages/core/parsers.adoc

```diff
@@ -1,8 +1,180 @@
 = Parsers
 
+Parsers are the core of what Pathom does. This section explains what Pathom parser does,
+explain their responsibilities and details how they work.
+
+== The parser job
+
+Every EQL transaction is a vector, this means that by definition an EQL transaction is
+a collection of things. The parser job is to walk this collection, figures the value
+of each requested entry and return a map with them.
+
+In case of Pathom parsers they also have a distinction between `read` and `mutate`, when
+the parser is built those are provided separate.
+
+I may look like a simple task, but it can get quite complicated depending on how you
+intend to coordinate that processing. Let's take a look at the built in parsers from
+Pathom:
+
 == Serial parser
 
-TODO: explain serial parser internals
+This is the simplest parser available, it doesn't do more than a reduce, calling the reader
+for each entry in the query.
+
+Just for improve understand, this is how we can implement a simple naive serial parser:
+
+[source,clojure]
+----
+; this a simple parser that reads entries from the env based on user query
+(defn reduce-parser [env eql]
+  (reduce
+    (fn [out entry]
+      (assoc out entry (get env entry)))
+    {}
+    eql))
+
+; process EQL!
+(reduce-parser {:foo "bar" :answer 42} [:answer])
+; => {:answer 42}
+----
+
+Now let's push a different query to our parser, adding a parameter to our entry:
+
+[source,clojure]
+----
+(reduce-parser {:foo "bar" :answer 42} ['(:answer {:with "param"})])
+; => {(:answer {:with "param"}) nil}
+----
+
+So that didn't go so well, we got now a key out that includes the list, and no value.
+
+We could deal with this manually, checking for the list, but EQL has enough syntax that
+this turns in trouble quickly. To fix this, instead of deal with the EQL directly, let's
```

Comment:
this turns into trouble quickly. To fix this, instead of dealing with EQL directly, let's